packages/flutter_tools/test/commands.shard/hermetic/version_test.dart

```diff
@@ -45,6 +45,27 @@ void main() {
       Pub: () => const Pub(),
     });
 
+    testUsingContext('version switch, latest commit query fails', () async {
+      const String version = '10.0.0';
+      final VersionCommand command = VersionCommand();
+      final Future<void> runCommand = createTestCommandRunner(command).run(<String>['version', version]);
+      await Future.wait<void>(<Future<void>>[runCommand]);
+      expect(testLogger.errorText, contains('git failed'));
+    }, overrides: <Type, Generator>{
+      ProcessManager: () => MockProcessManager(latestCommitFails: true),
+      Pub: () => const Pub(),
```

Comment:
Does this need the real pub? Or could it work if it supported --no-pub

Reply:
These tests don't need a real `Pub`. On investigating, I think the way that `version` and `upgrade` invoke pub should be revisited. I'll take a look in a separate PR.